src/web/src/components/decoration/PropPicker.svelte

```diff
@@ -12,114 +13,63 @@
 	import { scale } from "svelte/transition";
 	import currency from "../../store/currency";
 
-	const categories = [
-		{
-			name: "Bar",
-			value: "bar",
-		},
-		{
-			name: "Bathroom",
-			value: "bathroom",
-		},
-		{
-			name: "Bins",
-			value: "bins",
-		},
-		{
-			name: "Construction",
-			value: "construction",
-		},
-		{
-			name: "Electrical",
-			value: "electrical",
-		},
-		{
-			name: "Equipment",
-			value: "equipment",
-		},
-		{
-			name: "Garage",
-			value: "garage",
-		},
-		{
-			name: "Industrial",
-			value: "industrial",
-		},
-		{
-			name: "Interior",
-			value: "interior",
-		},
-		{
-			name: "Kitchen",
-			value: "kitchen",
-		},
-		{
-			name: "Minigame",
-			value: "minigame",
-		},
-		{
-			name: "Office",
-			value: "office",
-		},
-		{
-			name: "Outdoor",
-			value: "outdoor",
-		},
-		{
-			name: "Potted",
-			value: "potted",
-		},
-		{
-			name: "Recreational",
-			value: "recreational",
-		},
-		{
-			name: "Rubbish",
-			value: "rubbish",
-		},
-		{
-			name: "Seating",
-			value: "seating",
-		},
-		{
-			name: "Storage",
-			value: "storage",
-		},
-		{
-			name: "Utility",
-			value: "utility",
-		},
-		{
-			name: "Walls and fences",
-			value: "walls",
-		},
-	];
-
-	let props: Promise<PropType[]>;
-	let category: any;
-	let isVisible: boolean;
-	let selectedProp: PropType | null;
-
-	const fetchProps = async (category: string) => {
-		if (isEnvBrowser()) {
-			return new Promise((r) => {
-				r([{ category: "", id: "v_ret_gc_chair03", name: "Chair", price: 70 }]);
-			});
-		}
-
-		if (!category) return;
-		return fetchNui("getProps", category);
-	};
-
-	$: props = fetchProps(category?.value);
-
-	const selectProp = async (model: string) => {
+    let categories = [];
+    let props = [];
+    let category = null;
+    let isVisible = false;
+    let selectedProp = null;
+	
+	// Fetch categories from the server
+    async function fetchCategories() {
+        const cats = await fetchNui("getCategories");
+        if (cats) {
+            categories = JSON.parse(cats);
```

Comment:
I don't think json encoding is required here, I think NUI requests does support sending json like objects